src/include/miopen/graphapi/operationgraph_descriptor.hpp

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/graphapi/graphapi.hpp>
+#include <miopen/graphapi/opgraph.hpp>
+#include <miopen/miopen.h>
+
+#include <cstdint>
+#include <vector>
+
+namespace miopen {
+
+namespace graphapi {
+
+class BackendOperationGraphDescriptor : public BackendDescriptor
+{
+private:
+    miopenHandle_t mHandle = nullptr;
+    std::vector<miopenBackendDescriptor_t> mOps; // to return them in getAttribute
+    OpGraphBuilder mOpGraphBuilder;
+    OpGraph mOpGraph;
+    std::vector<miopenSolution_t> mSolutions;
```

Comment:
@atamazov @DrizztDoUrden @amberhassaan Can we use `miopen::Solution` here?
It seems like 'Engine' in graph API is the same as `miopen::Solution`. We need a C++ wrapper for `miopenSolution_t` and that C-interface of it. Isn't it better to directly use `miopen::Solution`?

Reply:
I'm of the opinion that we create a wrapper to have a name `Engine` that is compatible with the graph API. It also allows us to move away from `Solution` in the future, if the need arises.

Reply:
@scerzh 
> Can we use `miopen::Solution` here? It seems like `Engine` in graph API is the same as `miopen::Solution`.

Good question. I am also a bit curious that object handle from public API is used for internal needs. I would prefer holding actual objects by-value.

> We need a C++ wrapper for miopenSolution_t 

It seems like `miopen::Solution` provides the necessary functionality.

> and that C-interface of it

We have it in miopen.h already of course. But in graph API engines are always hidden behind various backend descriptors and function calls, so AFAICS we do not need any dedicated C API for it.

@amberhassaan 
> I'm of the opinion that we create a wrapper to have a name `Engine` that is compatible with the graph API.

AFAICS engines are not exposed in the graph API, pls. see my previous comment, thanks.

Reply:
@atamazov : we have plans to release graph API as a C++ API as well.

Reply:
@amberhassaan Ah, I see. Well, I thought we can fork https://github.com/NVIDIA/cudnn-frontend and replace their backend with ours. If this is acceptable (AFAICS their license allows that)...

Reply:
That is another approach too that needs more discussion. I, so far, don't like the idea of adding our backend to their frontend API, because we don't support the backend API in full yet and there's some stuff around engines, knobs etc., that does not make sense and we might not want to support it or we might not know how to support it.

Reply:
@amberhassaan a C++ class representing the descriptor and a builder for it have been added to support MHA forward C++ test